### PR TITLE
Make the default stack.yaml be for GHC 8.2.2

### PR DESCRIPTION
**Pull request by alanz**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@ dependencies:
     - stack upgrade
```
Comment (nponeccop) [inline circle.yml]: Why do we do `stack setup` 2 times (see the log).

File: circle.yml [hunk 2]
```diff
@@ -32,16 +34,16 @@ dependencies:
         timeout: 7200
 
     - stack clean
-    - stack --stack-yaml=stack-8.2.2.yaml setup
-    - stack --stack-yaml=stack-8.2.2.yaml build  -j 2 --test --only-dependencies :
+    - stack --stack-yaml=stack-8.0.2.yaml setup
+    - stack --stack-yaml=stack-8.0.2.yaml build  -j 2 --test --only-dependencies :
         timeout: 7200
-    - stack --stack-yaml=stack-8.2.2.yaml build -j 2 :
+    - stack --stack-yaml=stack-8.0.2.yaml build -j 2 :
```
Comment (nponeccop) [inline circle.yml]: This line doesn't make sense to me. Basically we build HIE just to clean it later and rebuild (and make 2x build time)

File: circle.yml [hunk 3]
```diff
@@ -50,6 +52,6 @@ test:
     - stack --stack-yaml=stack-8.2.1.yaml test -j 2  haskell-ide-engine :
         timeout: 14400
 
-    - stack --stack-yaml=stack-8.2.2.yaml clean
-    - stack --stack-yaml=stack-8.2.2.yaml test -j 2  haskell-ide-engine :
+    - stack --stack-yaml=stack-8.0.2.yaml clean
```
Comment (nponeccop) [inline circle.yml]: This one should be 8.2.1 - we clean the previous build

File: circle.yml (hunk 2 above)
Comment (alanz) [reply circle.yml]: The initial build is to get the deps installed, I think.

And I was getting builds stomping on each other, so perhaps went overboard on this stuff.

File: circle.yml (hunk 3 above)
Comment (alanz) [reply circle.yml]: ok

**Comment by alanz** [comment]
BTW, technically this PR is intended to just be a renaming of files, to bring a different default into scope for `stack.yaml`.

So perhaps we should keep it as that, and then clean it up.

**Comment by nponeccop** [comment]
> So perhaps we should keep it as that, and then clean it up.

I agree

**Pull request merged**